src/iexamine.cpp

```diff
@@ -2293,6 +2295,102 @@ void iexamine::kiln_full( player &, const tripoint &examp )
+    return it.has_flag( "FIRESTARTER" ) || it.has_flag( "FIRE" );
+    } ) ) {
+        const auto usef = it->type->get_use( "firestarter" );
+        if( usef != nullptr && usef->get_actor_ptr() != nullptr ) {
+            const auto actor = dynamic_cast<const firestarter_actor *>( usef->get_actor_ptr() );
+            if( actor->can_use( p, *it, false, examp ).success() ) {
+                firestarters.insert( std::pair<int, item *>( actor->moves_cost_fast, it ) );
+            }
+        }
+    }
+
+    const bool has_firestarter = firestarters.size() > 0;
+    const bool has_bionic_firestarter = p.has_bionic( bionic_id( "bio_lighter" ) ) &&
+                                        p.power_level >= bionic_id( "bio_lighter" )->power_activate;
+
+    uilist selection_menu;
+    selection_menu.text = _( "Select an action" );
+    selection_menu.addentry( 0, true, 'e', _( "Examine" ) );
+    if( !already_on_fire ) {
+        selection_menu.addentry( 1, has_firestarter, 'f',
+                                 has_firestarter ? "Start a fire" : "Start a fire... you'll need a fire source." );
+        if( has_bionic_firestarter ) {
+            selection_menu.addentry( 2, true, 'b', "Use a CBM to start a fire" );
+        }
+    }
+    if( furn_is_deployed ) {
+        selection_menu.addentry( 3, true, 't', string_format( _( "Take down the %s" ),
+                                 g->m.furnname( examp ) ) );
+    }
+    selection_menu.query();
+
+    switch( selection_menu.ret ) {
+        case 0:
+            none( p, examp );
+            Pickup::pick_up( examp, 0 );
+            return;
+        case 1: {
+            for( auto &firestarter : firestarters ) {
+                item *it = firestarter.second;
+                const auto usef = it->type->get_use( "firestarter" );
+                const auto actor = dynamic_cast<const firestarter_actor *>( usef->get_actor_ptr() );
+                p.add_msg_if_player( string_format( _( "You attempt to start a fire with your %s..." ),
+                                                    it->tname() ) );
+                const ret_val<bool> can_use = actor->can_use( p, *it, false, examp );
+                if( can_use.success() ) {
+                    const long charges = actor->use( p, *it, false, examp );
+                    p.use_charges( it->typeId(), charges );
+                    return;
+                } else {
+                    p.add_msg_if_player( m_bad, can_use.str() );
+                }
+            }
+            p.add_msg_if_player( _( "You weren't able to start a fire." ) );
+            return;
+        }
+        case 2: {
+            if( g->m.add_field( examp, fd_fire, 1 ) ) {
+                p.charge_power( -bionic_id( "bio_lighter" )->power_activate );
+                p.mod_moves( -100 );
+            } else {
+                p.add_msg_if_player( m_info, _( "You can't light a fire there." ) );
+            }
+            return;
+        }
+        case 3: {
```

Comment:
Is it possible to invoke `deployed_furniture` `iexamine` function here instead of copypasting it?

Reply:
It's possible, but it would add an extra key press and lose the check for existing fire.